### PR TITLE
Show logo and header colour as per current deployment env config

### PR DESCRIPTION
**Pull request by timmyjose**
PR for story [As an Admin, I should instantly know which environment I am on (prod, sandbox or staging)](https://www.pivotaltracker.com/n/projects/2184102/stories/163934813)

### CODE REVIEW

File: spec/helpers/simple_server_env_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe SimpleServerEnvHelper do
+  include SimpleServerEnvHelper
+
+  let(:simple_server_env) {"SIMPLE_SERVER_ENV"}
```
Comment (kitallis) [inline spec/helpers/simple_server_env_helper_spec.rb]: Just a styling thing: put spaces after opening and before closing curly brace. You can set this up in RubyMine to do automatically.

File: spec/helpers/simple_server_env_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  after(:each) do
+    ENV[simple_server_env] = "test"
+  end
+
+  describe 'style_class_for_environment' do
+    it 'should return the correct style class for the environment' do
+      ENV[simple_server_env] = "qa"
+
+      style = style_class_for_environment
+      puts style
```
Comment (kitallis) [inline spec/helpers/simple_server_env_helper_spec.rb]: Don't leave stray `puts` in the spec.

File: spec/helpers/simple_server_env_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+RSpec.describe SimpleServerEnvHelper do
+  include SimpleServerEnvHelper
+
+  let(:simple_server_env) {"SIMPLE_SERVER_ENV"}
+
+  after(:each) do
+    ENV[simple_server_env] = "test"
+  end
+
+  describe 'style_class_for_environment' do
```
Comment (kitallis) [inline spec/helpers/simple_server_env_helper_spec.rb]: Might be better to contextualize these specs within multiple `it`s per environment. Simpler to keep one assertion per spec.

File: spec/helpers/simple_server_env_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+      expect(style_class_for_environment).to eq 'navbar-sandbox'
+
+      ENV[simple_server_env] = "staging"
+
+      expect(style_class_for_environment).to eq 'navbar-staging'
+    end
+  end
+
+  describe 'logo_for_environment' do
```
Comment (kitallis) [inline spec/helpers/simple_server_env_helper_spec.rb]: One assertion per spec.

File: spec/helpers/simple_server_env_helper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      expect(logo_for_environment).to eq logo_for_sandbox
+
+      ENV[simple_server_env] = "staging"
+      logo_for_staging = image_tag 'simple_logo_staging.svg', width: 30, height: 30, class: "d-inline-block mr-2 align-top", alt: 'Simple Server Staging Logo'
+
+      expect(logo_for_environment).to eq logo_for_staging
+    end
+  end
+
+  describe 'alt_for_environment' do
```
Comment (kitallis) [inline spec/helpers/simple_server_env_helper_spec.rb]: One assertion per spec.

File: app/assets/stylesheets/application.scss [hunk 6]
```diff
@@ -751,3 +751,75 @@ nav.breadcrumb {
      font-family: $font-condensed;
 
   }
+
```
Comment (kitallis) [inline app/assets/stylesheets/application.scss]: Is `!important` truly necessary here? Ignore this comment if you think class precedence isn't enough to override these styles.

File: spec/helpers/simple_server_env_helper_spec.rb (hunk 1 above)
Comment (timmyjose) [reply spec/helpers/simple_server_env_helper_spec.rb]: Implemented changes.

File: spec/helpers/simple_server_env_helper_spec.rb (hunk 2 above)
Comment (timmyjose) [reply spec/helpers/simple_server_env_helper_spec.rb]: Removed the `puts` and the redundant var.

File: app/assets/stylesheets/application.scss (hunk 6 above)
Comment (timmyjose) [reply app/assets/stylesheets/application.scss]: Removed all the `!important` markers.

File: spec/helpers/simple_server_env_helper_spec.rb (hunk 3 above)
Comment (timmyjose) [reply spec/helpers/simple_server_env_helper_spec.rb]: Added contexts per environment.

File: spec/helpers/simple_server_env_helper_spec.rb (hunk 4 above)
Comment (timmyjose) [reply spec/helpers/simple_server_env_helper_spec.rb]: Changed to one assertion per spec.

File: spec/helpers/simple_server_env_helper_spec.rb (hunk 5 above)
Comment (timmyjose) [reply spec/helpers/simple_server_env_helper_spec.rb]: Changed to one assertion per spec.

**Comment by timmyjose** [comment]
Adding some screenshots of all possible environments post this change:

 * Default/ No specified environment/QA:

default_logged_in
default_logged_out
   
 * Production:

production_logged_in
production_logged_out

 * Staging:

staging_logged_in
staging_logged_out

 * Sandbox:

sandbox_logged_in
sandbox_logged_out

File: app/helpers/simple_server_env_helper.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
+module SimpleServerEnvHelper
+  SIMPLE_SERVER_ENV = "SIMPLE_SERVER_ENV".freeze
```
Comment (timcheadle) [inline app/helpers/simple_server_env_helper.rb]: This is not really necessary; just put the string in the `ENV.fetch()` lines below. If we have to change this ever, we'll have to grep the entire codebase anyway.

File: app/helpers/simple_server_env_helper.rb [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+  CUSTOMIZED_ENVS = %w[qa staging sandbox production].freeze
```
Comment (timcheadle) [inline app/helpers/simple_server_env_helper.rb]: We should include `development` in the list.

File: app/helpers/simple_server_env_helper.rb (hunk 7 above)
Comment (timmyjose) [reply app/helpers/simple_server_env_helper.rb]: Replaced variable with string in the `ENV.fetch` call itself.

File: app/helpers/simple_server_env_helper.rb (hunk 8 above)
Comment (timmyjose) [reply app/helpers/simple_server_env_helper.rb]: Added `development` to `CUSTOMIZED_ENVS`.

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -19,10 +19,10 @@
   </head>
```
[the remark this answers is not in the document]
Comment (timmyjose) [reply app/views/layouts/application.html.erb]: Added the current deployment env (qa, staging, sandbox, development) to the page title. For `production`, the title will be just `Simple Server`.

**Pull request merged**